toffy/streak_detection_test.py

```diff
@@ -0,0 +1,445 @@
+        )
+        return _test_data
+
+    return streak_data_generator
+
+
+@pytest.fixture(scope="function")
+def df_dataset():
+    def test_df_generator(shape: Tuple[int, int], df_row_num: int):
+        col_size = shape[0]
+        row_size = shape[1]
+
+        # Initialize a new generator - set seed for reproducibility
+        rng = np.random.default_rng(12345)
+
+        # Create test_df
+        min_row = rng.integers(low=0, high=row_size - 1, size=(df_row_num))
+        max_row = min_row + 1
+        min_col = rng.integers(low=0, high=col_size - 1, size=(df_row_num))
+        max_col = rng.integers(low=min_col, high=row_size - 1, size=(df_row_num))
+        test_df = pd.DataFrame(
+            np.stack([min_row, max_row, min_col, max_col], axis=-1),
+            columns=["min_row", "max_row", "min_col", "max_col"],
+        )
+        return test_df
+
+    return test_df_generator
+
+
+def test_get_save_dir(tmp_path):
+    data_dir: Path = Path(tmp_path)
+    name: str = "streak_data_test"
+    ext_csv: str = "csv"
+    ext_tiff: str = "tiff"
+
+    csv_save_dir: Path = sd._get_save_dir(data_dir=data_dir, name=name, ext=ext_csv)
+    assert csv_save_dir == Path(tmp_path / "streak_data_test.csv")
+
+    tiff_save_dir: Path = sd._get_save_dir(data_dir=data_dir, name=name, ext=ext_tiff)
+    assert tiff_save_dir == Path(tmp_path / "streak_data_test.tiff")
+
+
+def test_save(tmp_path, streak_dataset):
+    # Create minimum data needed for testing the save functionality with dataframes and numpy array
+    # streak_data needs: corrected_dir, streak_channel, and data (streak_mask, streak_df,
+    # filtered_streak_mask, filtered_streak_df, boxed_streaks, corrected_streak_mask)
+
+    streak_data_test = streak_dataset(
+        corrected_dir=tmp_path, fov="fov0", chan="chan0", shape=(1000, 1000)
+    )
+
+    Read_Fn = namedtuple("Read_Operation", "read_fn, ext")
+    fields = {
+        "streak_mask": Read_Fn(io.imread, ".tiff"),
+        "streak_df": Read_Fn(pd.read_csv, ".csv"),
+        "filtered_streak_mask": Read_Fn(io.imread, ".tiff"),
+        "filtered_streak_df": Read_Fn(pd.read_csv, ".csv"),
+        "boxed_streaks": Read_Fn(io.imread, ".tiff"),
+        "corrected_streak_mask": Read_Fn(io.imread, ".tiff"),
+    }
+
+    #  Test `streak_detection::_save`
+    for field, rf in fields.items():
+        sd._save(streak_data=streak_data_test, name=field)
+
+        if rf.ext == ".csv":
+            data = rf.read_fn(Path(tmp_path, f"streak_data_chan0", field + rf.ext), index_col=0)
+        else:
+            data = rf.read_fn(Path(tmp_path, f"streak_data_chan0", field + rf.ext))
+
+        if type(data) is np.ndarray:
+            assert np.array_equal(data, getattr(streak_data_test, field))
+        if type(data) is pd.DataFrame:
+            pd.testing.assert_frame_equal(data, getattr(streak_data_test, field))
+
+
+def test_save_streak_masks(tmp_path, streak_dataset):
+    # * Minimum Data Needed:
```

Comment:
For describing test functions, it's more useful to comment on the logic within the function, like why an assert must be true, instead of describing the test data itself